### ISSUE TITLE
Inconsistently finds `-DCMAKE_TOOLCHAIN_FILE` for vcpkg? (Windows vs macOS and Linux)

### ISSUE THREAD

**Issue by SamuelMarks** [lukka/run-vcpkg#99]
If you look at my GitHub Actions you'll see many failed attempts at getting this to work.
https://github.com/SamuelMarks/curl-simple-https/actions

Works with Linux and macOS:
- https://github.com/SamuelMarks/curl-simple-https/actions/runs/1180363495

Works with Windows:
- https://github.com/SamuelMarks/curl-simple-https/actions/runs/1225637879

Considering my project is simple—with just the one dependency: libcurl—why is this never working?

---

A collection of errors:

[Linux]
> fatal error: curl/curl.h: No such file or directory
>      8 | #include <curl/curl.h>
>        |          ^~~~~~~~~~~~~

[macOS]
>   ld: symbol(s) not found for architecture x86_64
>  clang: error: linker command failed with exit code 1 (use -v to see invocation)
>  ninja: build stopped: subcommand failed.

[Windows]
>   MSBUILD : error MSB1016: Specify the verbosity level.
>  Switch: -v
>  
>  For switch syntax, type "MSBuild -help"

---

Based off https://github.com/lukka/CppBuildTasks-Validation/blob/116611c/.github/workflows/hosted-basic.yml or https://github.com/lukka/CppBuildTasks-Validation/blob/116611c/.github/workflows/hosted-advanced.yml

---

What am I doing wrong?

**Comment by lukka** [comment]
@SamuelMarks i think the first step to fix is to use the release branches (e.g. `vX`) rather than the `main` branch. For example `run-cmake@v3` rather than `run-cmake@main`.
I understand that the samples are using `main` and this is my fault as I am pushing stuff directly onto `main` lately to test things out
.
Then other errors may come up, and those are the ones that makes more sense to investigate.

**Comment by SamuelMarks** [comment]
@lukka I've done that. This line should work, right?
```yaml
useVcpkgToolchainFile: true
```

Because neither Linux nor macOS variants are playing nice (whereas Windows works everytime… now). Locally on Linux, macOS, and Windows it builds fine. Just not with your GitHub Action.

**Comment by SamuelMarks** [comment]
@lukka bump

**Comment by lukka** [comment]
@SamuelMarks yes, as long as `VCPKG_ROOT` is defined as a environment variable by the previous `run-cmake` execution, the `run-vcpkg` action would consume its toolchain when the `useVcpkgToolchainFile` is `true`. Two requests:
- try to minimize a desired workflow using the sample provided in the ["pure" workflow](https://github.com/lukka/run-cmake/issues/41).
- your existing workflow is invoking `run-cmake` multiple times, I think that is not desirable. Try to minimize it to the strict necessary, and provide a run of it with [debug logging enabled](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging), so it would be much easier to follow what is going on.
